core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -733,6 +733,7 @@ protected IntSet getOwnedSegments(ConsistentHash consistentHash) {
       InternalMetadataImpl metadata = new InternalMetadataImpl(e);
       PutKeyValueCommand put = commandsFactory.buildPutKeyValueCommand(e.getKey(), e.getValue(), segmentId,
                                                                        metadata, STATE_TRANSFER_FLAGS);
+      put.setInternalMetadata(e.getInternalMetadata());
```

Comment:
We should add `MetaParamsInternalMetadata` to `CommandsFactory#buildPutKeyValueCommand`.

Similarly, I noticed that the internal metadata is not  set in `ProtobufMetadataManagerInterceptor`. How are internal commands handled by xsite, should this be set there as well?

Reply:
> We should add `MetaParamsInternalMetadata` to `CommandsFactory#buildPutKeyValueCommand`.
> 
I didn't add it because I want to try to avoid touching too much interface... it would require a lot of changes everywhere and most of the places, the value is `null`. 

> Similarly, I noticed that the internal metadata is not set in `ProtobufMetadataManagerInterceptor`.
Is it needed? and set it to what exactly? (sorry, no idea what ProtobufMetadataManagerInterceptor is)

The current xsite impl doesn't use it. 
In the new one, the internal metadata is set by the primary owner before replicating it to the backup owners. 
It is also used by the state transfer so the joiners can set the correct versions.